### PR TITLE
Updating to http-trigger templates to use a request body

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: Templates/HttpTrigger-CSharp/run.csx [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
-    var queryParamms = req.GetQueryNameValuePairs()
-        .ToDictionary(p => p.Key, p => p.Value, StringComparer.OrdinalIgnoreCase);
+    // parse query parameter
+    var queryParam = req.GetQueryNameValuePairs()
+                 .ToDictionary(p => p.Key, p => p.Value, StringComparer.OrdinalIgnoreCase);
 
     log.Verbose($"C# HTTP trigger function processed a request. RequestUri={req.RequestUri}");
 
+    // Get request body
+    string jsonContent = req.Content.ReadAsStringAsync().Result;
```
Comment (davidebbo) [inline Templates/HttpTrigger-CSharp/run.csx]: Did you check that this doesn't blow up on GET requests? Not saying it does, but worth making sure

Comment (soninaren) [inline Templates/HttpTrigger-CSharp/run.csx]: Yes, verified

File: Templates/HttpTrigger-CSharp/sample.dat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+	"name" : "Azure"
```
Comment (davidebbo) [inline Templates/HttpTrigger-CSharp/sample.dat]: AVoid using tab chars

File: Templates/HttpTrigger-NodeJS/sample.dat [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+	"name" : "Azure"
```
Comment (davidebbo) [inline Templates/HttpTrigger-NodeJS/sample.dat]: tabs

File: Templates/HttpTrigger-NodeJS/index.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
-﻿module.exports = function (context, req) {
-    context.log('Node.js HTTP trigger function processed a request. RequestUri=%s', req.originalUrl);
+﻿module.exports = function(context, req) {
+    context.log('Node.js HTTP trigger function processed a request. RequestUri:', req.originalUrl);
```
Comment (davidebbo) [inline Templates/HttpTrigger-NodeJS/index.js]: It seems strange to pass multiple params with no format string. Not intuitive what it even means.

Comment (soninaren) [inline Templates/HttpTrigger-NodeJS/index.js]: makes sense will add it back

File: Templates/HttpTrigger-CSharp/sample.dat (hunk 2 above)
Comment (soninaren) [inline Templates/HttpTrigger-CSharp/sample.dat]: fixed

File: Templates/HttpTrigger-NodeJS/sample.dat (hunk 3 above)
Comment (soninaren) [inline Templates/HttpTrigger-NodeJS/sample.dat]: fixed

File: Templates/HttpTrigger-NodeJS/index.js (hunk 4 above)
Comment (soninaren) [inline Templates/HttpTrigger-NodeJS/index.js]: fixed

**Comment by davidebbo** [comment]
:shipit:

File: Templates/HttpTrigger-CSharp/run.csx [hunk 5]
```diff
@@ -1,28 +1,42 @@
...
 using System.Net;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 
 public static Task<HttpResponseMessage> Run(HttpRequestMessage req, TraceWriter log)
 {
-    var queryParamms = req.GetQueryNameValuePairs()
-        .ToDictionary(p => p.Key, p => p.Value, StringComparer.OrdinalIgnoreCase);
+    // parse query parameter
+    var queryParam = req.GetQueryNameValuePairs()
```
Comment (mathewc) [inline Templates/HttpTrigger-CSharp/run.csx]: Why did you rename this to singular `queryParam`? It's a collection.

File: Templates/HttpTrigger-CSharp/run.csx (hunk 1 above)
Comment (mathewc) [inline Templates/HttpTrigger-CSharp/run.csx]: Recommend declaring the function async and using await rather than .Result
